fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java

```diff
@@ -468,13 +477,23 @@ public void refreshTable(HiveMetaStoreTableInfo hmsTable)
 
     public void refreshPartition(HiveMetaStoreTableInfo hmsTable, List<String> partNames) throws DdlException {
         GlobalStateMgr.getCurrentState().getMetastoreEventsProcessor().getEventProcessorLock().writeLock().lock();
+        Map<HivePartitionKey, Future<HivePartition>> hivePartitionFutures = new HashMap();
+        Map<HivePartitionKey, Future<HivePartitionStats>> hivePartitionStatsFutures = new HashMap();
         try {
             for (String partName : partNames) {
                 List<String> partValues = client.partitionNameToVals(partName);
                 HivePartitionKey key = new HivePartitionKey(hmsTable.getDb(), hmsTable.getTable(),
                         hmsTable.getTableType(), partValues);
-                partitionsCache.put(key, loadPartition(key));
-                partitionStatsCache.put(key, loadPartitionStats(key));
+                Future<HivePartition> partitionFuture = partitionRefreshExecutor.submit(() -> loadPartition(key));
+                Future<HivePartitionStats> partitionStatsFuture = partitionRefreshExecutor.submit(() -> loadPartitionStats(key));
+                hivePartitionFutures.put(key, partitionFuture);
+                hivePartitionStatsFutures.put(key, partitionStatsFuture);
+            }
+            for (Map.Entry<HivePartitionKey, Future<HivePartition>> entry : hivePartitionFutures.entrySet()) {
+                partitionsCache.put(entry.getKey(), entry.getValue().get());
+            }
```

Comment:
partition should be refreshed before partitionStats. you can refer to loadPartitionStats()

Reply:
> partition should be refreshed before partitionStats. you can refer to loadPartitionStats()

done